src/k2/cpo/client/CPOClient.h

```diff
@@ -259,6 +260,61 @@ class CPOClient {
         });
     }
 
+    template<typename ClockT=Clock>
+    seastar::future<std::tuple<Status, dto::MetadataLogStreamRegisterResponse>> RegisterMetadataStreamLog(Deadline<ClockT> deadline, String name, String plogId) {
+        dto::MetadataLogStreamRegisterRequest request{.name = std::move(name), .plogId=std::move(plogId)};
+
+        Duration timeout = std::min(deadline.getRemaining(), cpo_request_timeout());
+        return RPC().callRPC<dto::MetadataLogStreamRegisterRequest, dto::MetadataLogStreamRegisterResponse>(dto::Verbs::CPO_METADATA_LOG_STREAM_REGISTER, request, *cpo, timeout)
+        .then([this, &request, deadline] (auto&& result) {
+            auto& [status, k2response] = result;
+
+            if (deadline.isOver()) {
```

Comment:
callRPC already does this for you, so you can remove this whole if block. Same comment for the rest of these functions.

Reply:
fixed